modulegen/internal/dependabot/main.go

```diff
@@ -0,0 +1,70 @@
+package dependabot
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/testcontainers/testcontainers-go/modulegen/internal/context"
+)
+
+// Generator is a struct that contains the logic to generate the dependabot config file.
+type Generator struct{}
+
+// AddModule update dependabot with the new module
+func (g Generator) AddModule(ctx context.Context, tcModule context.TestcontainersModule) error {
+	configFile := ctx.DependabotConfigFile()
+
+	config, err := readConfig(configFile)
+	if err != nil {
+		return fmt.Errorf("read config: %w", err)
+	}
+
+	if err := config.addUpdate(string(filepath.Separator) + filepath.Join(tcModule.ParentDir(), tcModule.Lower())); err != nil {
+		return fmt.Errorf("add update: %w", err)
+	}
+
+	return writeConfig(configFile, config)
+}
+
+// Generate generates dependabot config file from source
+func (g Generator) Generate(ctx context.Context, examples []string, modules []string) error {
+	configFile := ctx.DependabotConfigFile()
+
+	config, err := readConfig(configFile)
+	if err != nil {
+		return fmt.Errorf("read config: %w", err)
+	}
+
+	for _, example := range examples {
+		if err := config.addUpdate(filepath.Join(string(filepath.Separator), "examples", example)); err != nil {
```

Comment:
question: this looks like its creating "/examples" or "\examples" which would break in windows if used what's the intent here?

Reply:
Mmm thinking carefully, dependabot file will always exist under our GitHub workers, so no need to run tests with different file separators, as they will always be linux